### PR TITLE
Add documentation and tests for label changelog label section assignment behavior

### PR DESCRIPTION
**Pull request by bnigh**
# What Changed

- Took attempt at updating label config docs to better communicate (a) default labels & (b) how PRs are assigned to a label section
- Added a few tests for changelog / releaseNote generation to check for documented behavior

# Why

- To improve accessibility of new auto users to understand what the default changelog / release notes will look like and how they might modify the config to achieve desired customization
- See [this comment](https://github.com/intuit/auto/issues/476#issuecomment-565922977) 

# Todo:

- [x] Add tests
- [x] Add docs

# Callouts

- I'm not 100% liking the way the changes to the docs turned out so far... but wanted to create a PR to get feedback / start the conversation of how to best improve the existing docs

Published PR with canary version: <details><summary>Published under canary scope @auto-canary</summary>
- `@auto-canary/auto@8.7.3-canary.822.10958.0`
- `@auto-canary/core@8.7.3-canary.822.10958.0`
- `@auto-canary/all-contributors@8.7.3-canary.822.10958.0`
- `@auto-canary/chrome@8.7.3-canary.822.10958.0`
- `@auto-canary/conventional-commits@8.7.3-canary.822.10958.0`
- `@auto-canary/crates@8.7.3-canary.822.10958.0`
- `@auto-canary/first-time-contributor@8.7.3-canary.822.10958.0`
- `@auto-canary/git-tag@8.7.3-canary.822.10958.0`
- `@auto-canary/jira@8.7.3-canary.822.10958.0`
- `@auto-canary/maven@8.7.3-canary.822.10958.0`
- `@auto-canary/npm@8.7.3-canary.822.10958.0`
- `@auto-canary/omit-commits@8.7.3-canary.822.10958.0`
- `@auto-canary/omit-release-notes@8.7.3-canary.822.10958.0`
- `@auto-canary/released@8.7.3-canary.822.10958.0`
- `@auto-canary/s3@8.7.3-canary.822.10958.0`
- `@auto-canary/slack@8.7.3-canary.822.10958.0`
- `@auto-canary/twitter@8.7.3-canary.822.10958.0`
- `@auto-canary/upload-assets@8.7.3-canary.822.10958.0`</details>

### CODE REVIEW

File: docs/pages/autorc.md [hunk 1]
```diff
@@ -107,6 +107,55 @@ To customize your project's labels use the `labels` section in your `.autorc`.
 }
 ```
 
+<details><summary>Click here to see the default label configuration</summary>
+
+```json
+[
```
Comment (hipstersmoothie) [inline docs/pages/autorc.md]: This is a great addition!

File: docs/pages/autorc.md [hunk 2]
```diff
@@ -132,16 +181,41 @@ You can customize everything about a label
...
+      "releaseType": "none"
+    }
+  ]
+}
+```
+
 #### Changelog Titles
 
-By default auto will create sections in the changelog for the following labels.
+Each PR included in the release will be matched to the first label defined in the config with a `changelogTitle` in priority order of the `releaseType` (major, minor, patch, then all others) and included as an entry within that section
```
[the remark this answers is not in the document]
Comment (bnigh) [reply docs/pages/autorc.md]: @hipstersmoothie 
I agree that the wording can be improved...

One piece of info that we may lose by removing that wording is the following scenario:

my config
```
{
  "labels": [
    { "releaseType": "none", "name": "foo", "changelogTitle": "Foo" },
    { "releaseType": "minor", "name": "bar", "changelogTitle": "Bar" }
  ]
}
```

if a PR has both `foo` and `bar` labels, then technically, the first label in the config is `foo`, but the PR would be assigned to the `bar` section.

This likely will not be a common use case, but wondering if it is worth keeping the info in the docs in a different format for completeness.

What are your thoughts?

**Comment by hipstersmoothie** [comment]
Ah I see. I just wanted to break up the sentence a little bit. I agree with
you we should keep the original wording maybe a little clearer

File: docs/pages/autorc.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (bnigh) [reply docs/pages/autorc.md]: > Ah I see. I just wanted to break up the sentence a little bit. I agree with
you we should keep the original wording maybe a little clearer

Definitely open to suggestions about how to improve the wording.

How about rewording to something like below?

---

#### Changelog Titles

Each PR included in the release will be assigned to a label section based upon the matching label with the highest `releaseType` that has a `changelogTitle`.
- Priority order of `releaseType` from highest to lowest: major, minor, patch, and then all others
- If a PR has multiple labels of the same `releaseType`, then the PR is assigned based upon the label that is assigned first in the config

By default auto will create sections in the changelog for the following labels:

- major
- minor
- patch
- internal
- documentation

For example:

- Using the default config, if a given PR has the labels `minor` and `internal`, then it will be included in the `minor` label section
- Using the default config, if a given PR has the labels `documentation` and `internal`, then it will be included in the `internal` label section

[the remark this answers is not in the document]
Comment (hipstersmoothie) [reply docs/pages/autorc.md]: I like it!

[the remark this answers is not in the document]
Comment (bnigh) [reply docs/pages/autorc.md]: should be addressed in: 0d286f9

[the remark this answers is not in the document]
Comment (bnigh) [reply docs/pages/autorc.md]: should be addressed in: 0d286f9

**Comment by adierkens** [comment]
:rocket: PR was released in v8.8.0 :rocket:
